### PR TITLE
[JSON-API] HOCON config json api

### LINKED ISSUE #11917: [JSON-API] Use HOCON config files instead of CLI parameters
As part of our transition to Daml 2.0 we need to change our configuration approach from CLI parameters to configuration files in order to align with the Canton's approach. We'll use https://github.com/lightbend/config and https://github.com/pureconfig/pureconfig to standardize around it. We will use the conventions established in Canton (https://www.canton.io/docs/stable/user-manual/usermanual/static_conf.html) to ensure consistency across our configurations.

below is a v1 config template based on existing CLI parameters.

```
http-json-api {
  ledger-api {
    address = "127.0.0.1"
    port = 5041
    tls {
      // the certificate to be used by the server
      cert-chain-file = "./tls/cert-chain.crt"
      // private key of the server
      private-key-file = "./tls/pvt-key.pem"
      // trust collection, which means that all client certificates will be verified using the trusted
      // certificates in this store. if omitted, the JVM default trust store is used.
      trust-collection-file = "./tls/root-ca.crt"
      // define whether clients need to authenticate as well (default not)
      client-auth = {
        // none, optional and require are supported
        type = require
        // If clients are required to authenticate as well, we need to provide a client
        // certificate and the key. 
        // If the server certificate is trusted by the trust-collection, then you can
        // just use the server certificates. Otherwise, you need to create separate ones.
        admin-client {
          cert-chain-file = "./tls/admin-client.crt"
          private-key-file = "./tls/admin-client.pem"
        }
      }
      // protocols = ...
      // ciphers = ...
    }
  }

  query-store {
    // can be postgres | oracle 
    type = postgres
    config {
        user = "postgres"
        password = "password"
        driver = org.postgresql.Driver
        url = "jdbc:postgresql://localhost:5432/test?&ssl=true"
    }
    // prefix for table names to avoid collisions, empty by default
    // table-prefix = ""
    // option setting how the schema should be handled. Valid options are create-only,start-only,create-if-needed-and-start,create-and-start.
    start-mode = "start-only"

    // max pool size for the database connection pool
    max-connections = 8
    //specifies the min idle connections for database connection pool.
    min-idle = 8
    //specifies the idle timeout for the database connection pool.
    idle-timeout = 10s
    / specifies the connection timeout for database connection pool.
    connection-timeout = 60s
  }

  
  //IP address that HTTP JSON API service listens on. Defaults to 127.0.0.1.
  address = "127.0.0.1"
  //HTTP JSON API service port number. A port number of 0 will let the system pick an ephemeral port.
  port = 8888
  //Optional unique file name where to write the allocated HTTP port number. If process terminates gracefully, this file will be deleted automatically. Used to inform clients in CI about which port HTTP JSON API listens on. Defaults to none, that is, no file gets created.
  port-file = "./port-file"

  // Optional interval to poll for package updates. Examples: 500ms, 5s, 10min, 1h, 1d. Defaults to 5 seconds
  package-reload-interval= 5s
  //Optional max inbound message size in bytes. Defaults to 4194304.
  max-inbound-message-size = 4194304 bytes     
  //Optional max inbound message size in bytes used for uploading and downloading package updates. Defaults to the `max-inbound-message-size` setting.       
  package-max-inbound-message-size = 4194304 bytes
  //Optional max cache size in entries for storing surrogate template id mappings. Defaults to None
  max-template-id-cache-entries = 1000   

  logging {
    //Default logging level to use. Available values are DEBUG, ERROR, TRACE, WARN, INFO. Defaults to INFO.
    level = "INFO"
    //which encoder to use: plain|json    
    log-encoder = "plain" 
  } 

  //Optional websocket configuration parameters
  websocket-config {
    //Maximum websocket session duration
    max-duration = 120m 
    //Server-side heartbeat interval duration
    heartbeat-period = 1s
  }
  
  metrics {
    //Start a metrics reporter. Must be one of "console", "csv:///PATH", "graphite://HOST[:PORT][/METRIC_PREFIX]", or "prometheus://HOST[:PORT]".
    reporter = "console"
    //Set metric reporting interval , examples : 1s, 30s, 1m, 1h
    reporting-interval = 30s
  }     

  // DEV MODE ONLY (not recommended for production)
  // Allow connections without a reverse proxy providing HTTPS.
  allow-insecure-tokens = false
  // Optional static content configuration string. Contains comma-separated key-value pairs, where:
  // prefix -- URL prefix,
  // directory -- local directory that will be mapped to the URL prefix.
  // Example: "prefix=static,directory=./static-content"
  // static-content {
  //    prefix = "static"
  //    directory = "./static-content"
  // }                              
}

```

### PR DESCRIPTION
**Pull request by akshayshirahatti-da**
Fixes #11917.

The new config would look like the below, all parameters except logging config and nonRepudiation settings can be supplied via the new HOCON config file. Logging and nonRepudiation config options are picked up via cli if provided.

```
{
  server {
    //IP address that HTTP JSON API service listens on. Defaults to 127.0.0.1.
    address = "127.0.0.1"
    //HTTP JSON API service port number. A port number of 0 will let the system pick an ephemeral port.
    port = 7500
    //Optional unique file name where to write the allocated HTTP port number. If process terminates gracefully, this file will be deleted automatically. Used to inform clients in CI about which port HTTP JSON API listens on. Defaults to none, that is, no file gets created.
    port-file = "port-file"
  }
  ledger-api {
    address = "127.0.0.1"
    port = 6400
    tls {
        enabled = "true"
        // the certificate to be used by the server
        cert-chain-file = "cert-chain.crt"
        // private key of the server
        private-key-file = "pvt-key.pem"
        // trust collection, which means that all client certificates will be verified using the trusted
        // certificates in this store. if omitted, the JVM default trust store is used.
        trust-collection-file = "root-ca.crt"
    }
  }

  query-store {
    base-config {
      user = "postgres"
      password = "password"
      driver = "org.postgresql.Driver"
      url = "jdbc:postgresql://localhost:5432/test?&ssl=true"

      // prefix for table names to avoid collisions, empty by default
      table-prefix = "foo"

      // max pool size for the database connection pool
      pool-size = 12
      //specifies the min idle connections for database connection pool.
      min-idle = 4
      //specifies the idle timeout for the database connection pool.
      idle-timeout = 12s
      //specifies the connection timeout for database connection pool.
      connection-timeout = 90s
    }
    // option setting how the schema should be handled.
    // Valid options are start-only, create-only, create-if-needed-and-start and create-and-start
    start-mode = "start-only"
    // any backend db specific values.
    backend-specific-conf {
      foo = "bar"
    }
  }



  // Optional interval to poll for package updates. Examples: 500ms, 5s, 10min, 1h, 1d. Defaults to 5 seconds
  package-reload-interval = 5s
  //Optional max inbound message size in bytes. Defaults to 4194304.
  max-inbound-message-size = 4194304
  //Optional max inbound message size in bytes used for uploading and downloading package updates. Defaults to the `max-inbound-message-size` setting.
  package-max-inbound-message-size = 4194304
  //Optional max cache size in entries for storing surrogate template id mappings. Defaults to None
  max-template-id-cache-entries = 2000
  //health check timeout
  health-timeout-seconds = 5

  //Optional websocket configuration parameters
  websocket-config {
    //Maximum websocket session duration
    max-duration = 180m
    //Server-side heartbeat interval duration
    heartbeat-period = 1s
    //akka stream throttle-mode one of either `shaping` or `enforcing`
    mode = "enforcing"
  }

  metrics {
    //Start a metrics reporter. Must be one of "console", "csv:///PATH", "graphite://HOST[:PORT][/METRIC_PREFIX]", or "prometheus://HOST[:PORT]".
    reporter = "console"
    //Set metric reporting interval , examples : 1s, 30s, 1m, 1h
    reporting-interval = 30s
  }

  // DEV MODE ONLY (not recommended for production)
  // Allow connections without a reverse proxy providing HTTPS.
  allow-insecure-tokens = false
  // Optional static content configuration string. Contains comma-separated key-value pairs, where:
  // prefix -- URL prefix,
  // directory -- local directory that will be mapped to the URL prefix.
  // Example: "prefix=static,directory=./static-content"
  static-content {
    prefix = "static"
    directory = "static-content-dir"
  }
}
```

-  [x] Update daml docs with new typeconf config

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/HttpJsonApiAppConfig.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      )
+    defined
+  }
+
+  val Empty: HttpJsonApiAppConfig = HttpJsonApiAppConfig(
+    HttpServerConfig(cliopts.Http.defaultAddress, -1),
+    LedgerApiConfig("", -1),
+  )
+}
+private[http] final case class HttpJsonApiAppConfig(
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/HttpJsonApiAppConfig.scala:62]:
```suggestion
private[http] final case class FileBasedConfig(
```

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import com.daml.pureconfigutils.SharedConfigReaders._
+import pureconfig.{ConfigReader, ConvertHelpers}
+import pureconfig.generic.semiauto._
+import ch.qos.logback.classic.{Level => LogLevel}
+
+import scala.concurrent.duration._
+
+private[http] object FileBasedConfig {
+  implicit val timeProviderTypeCfgReader: ConfigReader[ThrottleMode] =
+    ConfigReader.fromString[ThrottleMode](ConvertHelpers.catchReadError { s =>
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala:21]: I'm not a fan of this function; addressed in 919a9938d4757c832c9c4fd8f102bbd209e73c75 in branch `hocon_json_api+s11`.

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      s.toLowerCase() match {
+        case "enforcing" => ThrottleMode.Enforcing
+        case "shaping" => ThrottleMode.Shaping
+        case s =>
+          throw new IllegalArgumentException(
+            s"Value '$s' for throttle-mode is not one of 'shaping' or 'enforcing'"
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala:27]: The error produced already reports `s` so we should not repeat it in this part of the message. Also addressed in the aforementioned commit.

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      queryStore,
+      staticContent,
+      allowInsecureTokens,
+      websocketConfig,
+      nonRepudiation,
+      logLevel,
+      logEncoder,
+      metricsReporter,
+      metrics.map(_.reportingInterval).getOrElse(metricsReportingInterval),
+      maxTemplateIdCacheEntries,
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/FileBasedConfig.scala:96]: Added named arguments in `hocon_json_api+s11` 4bc25c05d39ea5a36b3d4a6a5081a0a93a21fb8f.

**Review by S11001001 — COMMENTED** [review]
Looks good; defining a fresh `case class` is nicer than defining an hlist type, after all.

There are probably some things we can report earlier by refining their types in `FileBasedConfig`, e.g. these various `Int`s could be newtyped (with appropriate config readers) so their problems get reported as part of config parsing instead of at some random time later, but this is perfectly reasonable as a port.

As mentioned I put my suggestions in the branch `hocon_json_api+s11`; you can simply merge that into this branch if you like.

**Comment by akshayshirahatti-da** [comment]
@S11001001 I have merged your changes in and tried to cleanup a bit based off that as well, please have a look and review if things look good overall. I'll make changes to daml docs once we are happy with these changes. /cc @stefanobaghino-da

File: ledger-service/http-json/src/test/resources/http-json-api.conf [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  allow-insecure-tokens = false
+  // Optional static content configuration string. Contains comma-separated key-value pairs, where:
+  // prefix -- URL prefix,
+  // directory -- local directory that will be mapped to the URL prefix.
+  // Example: "prefix=static,directory=./static-content"
+  static-content {
+    prefix = "static"
+    directory = "static-content-dir"
+  }
+}
```
Comment (akshayshirahatti-da) [inline ledger-service/http-json/src/test/resources/http-json-api.conf:94]: need newline

File: ledger-service/http-json/src/test/resources/http-json-api-minimal.conf [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  server {
+    address = "127.0.0.1"
+    port = 7500
+  }
+  ledger-api {
+    address = "127.0.0.1"
+    port = 6400
+  }
+}
```
Comment (akshayshirahatti-da) [inline ledger-service/http-json/src/test/resources/http-json-api-minimal.conf:10]: need newline

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala [hunk 7]
```diff
@@ -6,7 +6,6 @@ package com.daml.http
 import java.io.File
 import java.nio.file.Path
 import java.util.concurrent.TimeUnit
-
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:9]: Undo it undo it!

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala [hunk 8]
```diff
@@ -15,9 +14,16 @@ import scalaz.Show
 import scalaz.StateT.liftM
 
 import scala.concurrent.duration._
-
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:18]: Undo it undo it!

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala [hunk 9]
```diff
@@ -15,9 +14,16 @@ import scalaz.Show
...
 import ch.qos.logback.classic.{Level => LogLevel}
 import com.daml.cliopts.Logging.LogEncoder
+import com.daml.http.WebsocketConfig.{
+  DefaultHeartbeatPeriod,
+  DefaultMaxBurst,
+  DefaultMaxDuration,
+  DefaultThrottleElem,
+  DefaultThrottleMode,
+  DefaultThrottlePer,
+}
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:26]:
```suggestion
import com.daml.http.{WebsocketConfig => WSC}
```

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala [hunk 10]
```diff
@@ -42,41 +48,38 @@ private[http] final case class Config(
...
-    throttlePer: FiniteDuration,
-    maxBurst: Int,
-    mode: ThrottleMode,
-    heartBeatPer: FiniteDuration,
+    maxDuration: FiniteDuration = DefaultMaxDuration,
+    throttleElem: Int = DefaultThrottleElem,
+    throttlePer: FiniteDuration = DefaultThrottlePer,
+    maxBurst: Int = DefaultMaxBurst,
+    mode: ThrottleMode = DefaultThrottleMode,
+    heartbeatPeriod: FiniteDuration = DefaultHeartbeatPeriod,
```
Comment (S11001001) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:66]:
```suggestion
    maxDuration: FiniteDuration = WSC.DefaultMaxDuration,
    throttleElem: Int = WSC.DefaultThrottleElem,
    throttlePer: FiniteDuration = WSC.DefaultThrottlePer,
    maxBurst: Int = WSC.DefaultMaxBurst,
    mode: ThrottleMode = WSC.DefaultThrottleMode,
    heartbeatPeriod: FiniteDuration = WSC.DefaultHeartbeatPeriod,
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 11]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
 
+  "TypeConfig app Conf" - {
+    val confFile = "ledger-service/http-json/src/test/resources/http-json-api-minimal.conf"
+
+    "should fail on missing ledgerHost and ledgerPort if no config file supplied" in {
+      configParser(sharedOptions.drop(4)) match {
+        case Some(_) =>
+          fail("Expected failure on missing required values --ledger-host and --ledger-port")
+        case None => succeed
+      }
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:236]:
```suggestion
      configParser(sharedOptions.drop(4)) should ===(None)
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 12]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
+    }
+    "should fail on missing httpPort and no config file is supplied" in {
+      configParser(sharedOptions.take(4)) match {
+        case Some(_) => fail("Expected failure on missing required value --http-port")
+        case None => succeed
+      }
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:242]: Likewise.

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 13]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
+    }
+
+    "should successfully load a minimal config file" in {
+      val cfg = configParser(Seq("--config", requiredResource(confFile).getAbsolutePath)).getOrElse(
+        fail(s"Unexpected failure parsing $confFile")
+      )
+      cfg shouldBe Config.Empty.copy(httpPort = 7500, ledgerHost = "127.0.0.1", ledgerPort = 6400)
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:249]:
```suggestion
      val cfg = configParser(Seq("--config", requiredResource(confFile).getAbsolutePath))
      cfg shouldBe Some(Config.Empty.copy(httpPort = 7500, ledgerHost = "127.0.0.1", ledgerPort = 6400))
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 14]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
+      ).getOrElse(
+        fail(s"Unexpected failure parsing $confFile")
+      )
+      cfg shouldBe Config(
+        httpPort = 7500,
+        ledgerHost = "127.0.0.1",
+        ledgerPort = 6400,
+        logLevel = Some(LogLevel.DEBUG),
+        logEncoder = LogEncoder.Json,
+      )
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:271]:
```suggestion
      )
      cfg shouldBe Some(Config(
        httpPort = 7500,
        ledgerHost = "127.0.0.1",
        ledgerPort = 6400,
        logLevel = Some(LogLevel.DEBUG),
        logEncoder = LogEncoder.Json,
      ))
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 15]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
+    }
+
+    "should fail when config file and cli args both are supplied" in {
+      configParser(
+        Seq("--config", requiredResource(confFile).getAbsolutePath) ++ sharedOptions
+      ) match {
+        case Some(_) => fail("Expected failure on supplying both config file and cli args ")
+        case None => succeed
+      }
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:280]:
```suggestion
      ) should ===(None)
```

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala [hunk 16]
```diff
@@ -216,4 +225,107 @@ final class CliSpec extends AnyFreeSpec with Matchers {
...
+        metricsReportingInterval = 30.seconds,
+        wsConfig = Some(expectedWsConfig),
+        surrogateTpIdCacheMaxEntries = Some(2000L),
+        packageMaxInboundMessageSize = Some(StartSettings.DefaultMaxInboundMessageSize),
+      )
+      val confFile = "ledger-service/http-json/src/test/resources/http-json-api.conf"
+      val cfg = configParser(Seq("--config", requiredResource(confFile).getAbsolutePath)).getOrElse(
+        fail(s"Unexpected failure parsing $confFile")
+      )
+      cfg shouldBe expectedConfig
```
Comment (S11001001) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:327]:
```suggestion
      val cfg = configParser(Seq("--config", requiredResource(confFile).getAbsolutePath))
      cfg shouldBe Some(expectedConfig)
```

File: ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala [hunk 17]
```diff
@@ -6,63 +6,129 @@ package com.daml.pureconfigutils
...
+final case class LedgerApiConfig(
+    address: String,
+    port: Int,
+    tls: LedgerTlsConfig = LedgerTlsConfig(),
+)
+final case class MetricsConfig(reporter: MetricsReporter, reportingInterval: FiniteDuration)
+
+object TokenVerifierConfig {
+  val knownTokenVerifiers =
+    List("hs256-unsafe", "rs256-crt", "es256-crt", "es512-crt", "rs256-jwks")
```
Comment (S11001001) [inline ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala:48]: I really like this idea, but I want to take it further to guarantee consistency between the error message and the actual set of verifiers, and also to not treat the first element of this list magically. That is done in 71680e16a251cffd32556822e97610c93078c905 in branch `hocon_json_api+s11`.

File: ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala [hunk 18]
```diff
@@ -6,63 +6,129 @@ package com.daml.pureconfigutils
...
       s.toLowerCase() match {
-        case "static" => TimeProviderType.Static
-        case "wall-clock" => TimeProviderType.WallClock
-        case s =>
-          throw new IllegalArgumentException(
-            s"Value '$s' for time-provider-type is not one of 'static' or 'wall-clock'"
-          )
+        case "static" => Right(TimeProviderType.Static)
+        case "wall-clock" => Right(TimeProviderType.WallClock)
+        case _ => Left("value  is not one of 'static' or 'wall-clock'")
```
Comment (S11001001) [inline ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala:115]:
```suggestion
        case _ => Left("not one of 'static' or 'wall-clock'")
```

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala (hunk 10 above)
Comment (S11001001) [reply ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:66]: c8643486acd

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala (hunk 9 above)
Comment (S11001001) [reply ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:26]: c8643486acd

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala (hunk 8 above)
Comment (S11001001) [reply ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:18]: 0b4ef48ffff

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala (hunk 7 above)
Comment (S11001001) [reply ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:9]: 0b4ef48ffff

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 11 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:236]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 12 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:242]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 13 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:249]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 14 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:271]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 15 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:280]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala (hunk 16 above)
Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/CliSpec.scala:327]: a477aeae3a16c8510ac6b52fe48b74ef3f99db11

File: ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala (hunk 18 above)
Comment (S11001001) [reply ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala:115]: 2afaebe4de738ad0de950fe25a488f2385733264

**Pull request merged**